OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/activities/NotificationOpenedActivityBase.kt

```diff
@@ -36,32 +36,26 @@ import com.onesignal.notifications.internal.open.INotificationOpenedProcessor
 abstract class NotificationOpenedActivityBase : Activity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        if (!OneSignal.initWithContext(applicationContext)) {
-            return
-        }
-
-        var self = this
-
-        suspendifyOnThread {
-            var openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
-            openedProcessor.processFromContext(self, intent)
-        }
-
-        finish()
+        processIntent()
     }
 
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
+        processIntent()
+    }
+
+    private fun processIntent() {
         if (!OneSignal.initWithContext(applicationContext)) {
             return
         }
-
-        var self = this
         suspendifyOnThread {
-            var openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
-            openedProcessor.processFromContext(self, getIntent())
+            val openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
+            openedProcessor.processFromContext(this, intent)
```

Comment:
Will "this" always carry the correct context of the NotificationOpenedActivityBase? It looks like suspendifyOnMain executes the block with context of Dispatchers.Main.

Reply:
@jinliu9508 Yes, we get the correct `Activity` type here.
![image](https://github.com/OneSignal/OneSignal-Android-SDK/assets/645861/590519e5-da4b-418f-9ea1-c04efdbe17a7)